instrumentation/rocketmq/src/test/java/brave/rocketmq/clients/ITRocketMQ.java

```diff
@@ -0,0 +1,114 @@
+ * the License.
+ */
+package brave.rocketmq.clients;
+
+import brave.messaging.MessagingRequest;
+import brave.messaging.MessagingTracing;
+import brave.sampler.Sampler;
+import brave.sampler.SamplerFunction;
+import brave.sampler.SamplerFunctions;
+import brave.test.ITRemote;
+import brave.test.IntegrationTestSpanHandler;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.File;
+import java.net.URL;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+
+class ITRocketMQ extends ITRemote {
+
+  static final DockerImageName IMAGE = DockerImageName.parse("apache/rocketmq:4.6.0");
+  static final String TOPIC_PREFIX = "JoeKerouac_Test_";
+
+  static final int NAMESERVER_PORT = 9876;
+  static final int BROKER_PORT = 10911;
+
+  static RocketMQContainer nameserver;
+  static RocketMQContainer broker;
+
+  public IntegrationTestSpanHandler producerSpanHandler = new IntegrationTestSpanHandler();
+  public IntegrationTestSpanHandler consumerSpanHandler = new IntegrationTestSpanHandler();
+
+  SamplerFunction<MessagingRequest> producerSampler = SamplerFunctions.deferDecision();
+  SamplerFunction<MessagingRequest> consumerSampler = SamplerFunctions.deferDecision();
+
+  RocketMQTracing producerTracing =
+    RocketMQTracing.create(MessagingTracing
+      .newBuilder(
+        tracingBuilder(Sampler.ALWAYS_SAMPLE).localServiceName("producer").clearSpanHandlers()
+          .addSpanHandler(producerSpanHandler).build())
+      .producerSampler(r -> producerSampler.trySample(r)).build());
+
+  RocketMQTracing consumerTracing =
+    RocketMQTracing.create(MessagingTracing
+      .newBuilder(
+        tracingBuilder(Sampler.ALWAYS_SAMPLE).localServiceName("consumer").clearSpanHandlers()
+          .addSpanHandler(consumerSpanHandler).build())
+      .consumerSampler(r -> consumerSampler.trySample(r)).build());
+
+  static final class RocketMQContainer extends GenericContainer<RocketMQContainer> {
+
+    RocketMQContainer(String command, int... exportPorts) {
+      super(IMAGE);
+      if (exportPorts != null && exportPorts.length > 0) {
+        List<String> portBindings = new ArrayList<>();
+        for (int exportPort : exportPorts) {
+          portBindings.add(String.format("%d:%d", exportPort, exportPort));
+        }
+        setPortBindings(portBindings);
+      }
+
+      // do not publish all ports
+      withCreateContainerCmdModifier(cmd -> {
+        if (cmd.getHostConfig() != null) {
+          cmd.getHostConfig().withPublishAllPorts(false);
+        }
+      });
+
+      setCommand(command);
+      this.waitStrategy =
+        Wait.forLogMessage(".*The.*boot success.*", 1).withStartupTimeout(Duration.ofSeconds(60));
+    }
+  }
+
+  @BeforeAll
```

Comment:
per other note.. use `@Container` to manage these

Reply:
I have already resolved the above issue. Please help review

Reply:
sorry maybe I wasn't clear. I meant literally use `@Container` and like how it is done in https://github.com/openzipkin/brave/blob/master/instrumentation/kafka-clients/src/test/java/brave/kafka/clients/ITKafkaTracing.java

once this is done and the other comments including removing spring are, I'll review again, but not before. that way I don't need to repeat existing comments and you can have focus on what's needed to land this.

One thing is that once this is in, we'll likely have to publish our own image for rocketmq to avoid docker.io rate limiting, but I can help with that

Reply:
@codefromthecrypt  I still don't quite understand your meaning. Regarding the use of `@Container`, I have already replaced `@BeforeAll` with `@Container` in the test cases. As for removing spring, are you referring to removing the spring-boot-rocketmq module?

Reply:
we currently do not have any code for spring boot, only minimal properties for some libraries whose primary abstraction is spring (spring-rabbit), so adding spring boot to this is more of a decision (e.g. why not micrometer/tracing, vs sleuth etc) Since all brave is low-level where possible, it is easy to make a "go" decision on the main rocketmq even if we don't put autoconfigure into this repo. Adding autoconfig here is adding more work forever to the maintainers and possibly in conflict with some other projects.

on testcontainers, it appears your last commit did what I mentioned.

So, let's remove the spring boot stuff from this PR and try to land it. If your PR is marked to allow edits by maintainers, I can help with small things after you remove the spring boot. sound ok?

Reply:
@codefromthecrypt I have removed the Spring part. and the PR has been marked to allow edits by maintainers.